src/sentry/api/helpers/group_index/update.py

```diff
@@ -300,6 +300,34 @@ def update_groups(
             res_type = GroupResolution.Type.in_next_release
             res_type_str = "in_next_release"
             res_status = GroupResolution.Status.pending
+        elif status_details.get("inUpcomingRelease"):
+            if len(projects) > 1:
+                return Response(
+                    {"detail": "Cannot set resolved in next release for multiple projects."},
+                    status=400,
+                )
+            release = (
+                status_details.get("inUpcomingRelease")
+                or Release.objects.filter(
+                    projects=projects[0], organization_id=projects[0].organization_id
+                )
+                .extra(select={"sort": "COALESCE(date_released, date_added)"})
+                .order_by("-sort")[0]
```

Comment:
what does this mean semantically, and is this still the correct behavior for "resolve in upcoming release"? (the case where there is no release passed in.

Reply:
this query is used in a couple places to find the most recent release for a project, which we still need for this because in `clear_expired_resolutions` we compare the newly created release and check if there are any `GroupResolution` objects that have a release for a matching project that is older. if there is, then we know a new release for that project is created, so we can clear the pending GroupResolution and say the Group is resolved.

Reply:
got it